src/Microsoft.AspNetCore.Server.IISIntegration/Server/IO/WebSocketsAsyncIOEngine.Initialize.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration
+{
+    internal partial class WebSocketsAsyncIOEngine
+    {
+        internal class AsyncInitializeOperation : AsyncIOOperation
+        {
+            private readonly WebSocketsAsyncIOEngine _engine;
+
+            private IntPtr _requestHandler;
+
+            public AsyncInitializeOperation(WebSocketsAsyncIOEngine engine)
+            {
+                _engine = engine;
+            }
+
+            public void Initialize(IntPtr requestHandler)
+            {
+                _requestHandler = requestHandler;
+            }
+
+            public override bool InvokeOperation()
+            {
+                var hr = NativeMethods.HttpFlushResponseBytes(_requestHandler, out var fCompletionExpected);
```

Comment:
This feels a bit overengineered for a single operation. Not against it, but it is a lot of code just to call HttpFlushResponseBytes.

Reply:
Kinda, but it still has to be awaitable and queuable